platform-metrics/src/test/java/org/hypertrace/core/serviceframework/metrics/PlatformMetricsRegistryTest.java

```diff
@@ -151,9 +155,70 @@ public void testDistributionSummary() {
     assertEquals(1, distribution.count());
     assertEquals(100, distribution.totalAmount());
     assertTrue(
-        Arrays.stream(distribution.takeSnapshot().percentileValues()).map(m -> m.percentile())
-            .collect(
-                Collectors.toList()).containsAll(List.of(0.5, 0.95, 0.99)));
+        Arrays.stream(distribution.takeSnapshot().percentileValues())
+            .map(m -> m.percentile())
+            .collect(Collectors.toList())
+            .containsAll(List.of(0.5, 0.95, 0.99)));
+  }
+
+  @Test
+  public void testCache() throws Exception {
+    initializeCustomRegistry(List.of("testing"));
+    Cache<String, Integer> cache = CacheBuilder.newBuilder().maximumSize(10).recordStats().build();
+    PlatformMetricsRegistry.registerCache("my.cache", cache);
+    Callable<Integer> loader =
+        new Callable<Integer>() {
+          @Override
+          public Integer call() throws Exception {
+            return -1;
+          }
+        };
+
+    // Doing some cache activity
+    cache.put("One", 1);
+    cache.put("Two", 2);
+    cache.get("One", loader);
+    cache.get("Two", loader);
+    cache.get("Two", loader);
+    cache.get("Three", loader);
+    cache.get("Failed", loader);
+
+    // expected hit count = 3.0, miss rate = 2.0
+
+    // Checking Cache Stats from registry
+    double hits = 0, misses = 0;
+    List<Meter> meterList = PlatformMetricsRegistry.getMeterRegistry().getMeters();
+    // Fetching the hits, misses registered in the registry by browsing through all the registered
+    // meters
+    for (int i = 0; i < meterList.size(); i++) {
```

Comment:
nit: You can use the enhanced for loop

Reply:
Actually, just realized there is a better way to search for the meters. You can use something like 
```
PlatformMetricsRegistry.getMeterRegistry().get("cache.gets").tag("cache", "my.cache").tag("result", "hit").meter()
```

Walk through the GuavaCacheMetrics.monitor method till io.micrometer.core.instrument.binder.cache.CacheMeterBinder#bindTo

That gives you the list of all the meters, gauges registered and the tags applied on them.

Reply:
Done.